### PR TITLE
LIME-1287 Added context to audit session item

### PR DESCRIPTION
**Pull request by SarahHillGDS**
### What changed
(PR-rebased)
Added context to sessionHandler.

### Why did it change

So that IPV_DL_CRI_START Event contains Context Field (if present)

### Issue tracking

- [LIME-1287](https://govukverify.atlassian.net/browse/LIME-1287)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[LIME-1287]: https://govukverify.atlassian.net/browse/LIME-1287?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ElliotMurphyGDS — APPROVED** [review]

File: session/src/main/java/uk/gov/di/ipv/cri/common/api/handler/SessionHandler.java [hunk 1]
```diff
@@ -30,9 +33,7 @@
 import uk.gov.di.ipv.cri.common.library.util.EventProbe;
 
 import java.time.Clock;
-import java.util.Map;
-import java.util.TreeMap;
-import java.util.UUID;
+import java.util.*;
```
Comment (jkunle) [inline session/src/main/java/uk/gov/di/ipv/cri/common/api/handler/SessionHandler.java:36]: Please could you adjust see:  https://gds-way.digital.cabinet-office.gov.uk/manuals/programming-languages/java.html#imports

File: session/src/main/java/uk/gov/di/ipv/cri/common/api/handler/SessionHandler.java [hunk 2]
```diff
@@ -123,9 +124,20 @@ public APIGatewayProxyResponseEvent handleRequest(
...
+                evidenceList.add(evidence);
+                auditService.sendAuditEvent(
+                        AuditEventType.START,
+                        new AuditEventContext(input.getHeaders(), auditSessionItem),
+                        new AuditEventExtensions(evidenceList));
+            } else {
+                auditService.sendAuditEvent(
+                        AuditEventType.START,
+                        new AuditEventContext(input.getHeaders(), auditSessionItem));
+            }
```
Comment (jkunle) [inline session/src/main/java/uk/gov/di/ipv/cri/common/api/handler/SessionHandler.java:140]: Maybe we could extract this into a method something like 

```
public void sendStartAuditEvent(SessionRequest sessionRequest) {
 List<Evidence> evidenceList = new ArrayList<>();
        if (!StringUtils.isBlank(sessionRequest.getContext())) {
            Evidence evidence = new Evidence();
            evidence.setContext(sessionRequest.getContext());
            evidenceList.add(evidence);
        }

        auditService.sendAuditEvent(
            AuditEventType.START,
            new AuditEventContext(headers, auditSessionItem),
            evidenceList.isEmpty() ? null : new AuditEventExtensions(evidenceList)
        );
}
```

File: session/src/main/java/uk/gov/di/ipv/cri/common/api/handler/SessionHandler.java (hunk 1 above)
Comment (SarahHillGDS) [reply session/src/main/java/uk/gov/di/ipv/cri/common/api/handler/SessionHandler.java]: Removed wildcard

File: session/src/main/java/uk/gov/di/ipv/cri/common/api/handler/SessionHandler.java (hunk 2 above)
Comment (SarahHillGDS) [reply session/src/main/java/uk/gov/di/ipv/cri/common/api/handler/SessionHandler.java]: Extracted method outside of handle request

**Comment by jkunle** [comment]
LGTM

**Review by jkunle — APPROVED** [review]

**Review by ElliotMurphyGDS — APPROVED** [review]

**Pull request merged**